waku/v2/protocol/waku_store/waku_store.nim

```diff
@@ -459,28 +459,65 @@ proc query*(w: WakuStore, query: HistoryQuery, handler: QueryHandlerFunc) {.asyn
   waku_store_messages.set(response.value.response.messages.len.int64, labelValues = ["retrieved"])
   handler(response.value.response)
 
+proc queryFrom*(w: WakuStore, query: HistoryQuery, handler: QueryHandlerFunc, peer: PeerInfo): Future[Result[int64, string]] {.async.} =
```

Comment:
I'd consider pulling out the Result[int64, string] as a separate type, such as `QueryResult` or similar. Minor detail though!

Reply:
Sure, addressed!